src/component/back_btn.rs

```diff
@@ -1,16 +1,12 @@
-use leptos::{component, expect_context, view, Callback, IntoView};
+use gloo::history::{BrowserHistory, History};
+use leptos::{component, view, Callback, IntoView};
 use leptos_icons::Icon;
-use leptos_router::use_navigate;
-
-use crate::state::history::HistoryCtx;
 
 #[component]
-pub fn BackButton(fallback: String) -> impl IntoView {
-    let history_ctx = expect_context::<HistoryCtx>();
-
+pub fn BackButton() -> impl IntoView {
     let go_back = Callback::new(move |_| {
-        let back_url = history_ctx.back(&fallback);
-        use_navigate()(&back_url, Default::default());
+        let history = BrowserHistory::new();
```

Comment:
how do we handle the fallback case?
If the user navigates from google.com to, say /refer-earn
where will they be redirected?

Reply:
There isn't a way without using context and if we use context it ignores navigateOptions passed there is no way to track that

Reply:
Do you think we should raise an issue with leptos that`use_location` should also return Navigate options

Reply:
It is not possible to even  use that. Callback does not give any params to determine if the navigated url replaced the existing stack.